plugins/source/github/resources/services/actions/workflow_runs.go

```diff
@@ -31,11 +32,29 @@ func fetchWorkflowRuns(ctx context.Context, meta schema.ClientMeta, _ *schema.Re
 	repo := c.Repository
 	listOpts := &github.ListOptions{PerPage: 100}
 	actionOpts := &github.ListWorkflowRunsOptions{ListOptions: *listOpts}
+
+	if c.Spec.TableOptions.WorkflowRuns.ParsedTimeSince != "" {
+		actionOpts.Created = ">=" + c.Spec.TableOptions.WorkflowRuns.ParsedTimeSince
+	}
+	var lastCreatedAt string
 	for {
 		workflowRuns, resp, err := c.Github.Actions.ListRepositoryWorkflowRuns(ctx, *repo.Owner.Login, *repo.Name, actionOpts)
 		if err != nil {
 			return err
 		}
+		// When setting created_at, the API will return a maximum of 1000 records, so if we reached the limit we need to get the other records
+		// Workflows runs are sorted by created_at in descending order, so we need to scope down the query
+		if len(workflowRuns.WorkflowRuns) == 0 && actionOpts.Page > 0 && actionOpts.Created != "" {
+			actionOpts.Page = 0
+			actionOpts.Created = c.Spec.TableOptions.WorkflowRuns.ParsedTimeSince + ".." + lastCreatedAt
```

Comment:
shouldn't we use `lastCreatedAt` value on the 2nd loop iteration & so on (`lastCreatedAt` + ".." + time.Now()...`)?

Reply:
> shouldn't we use `lastCreatedAt` value on the 2nd loop iteration & so on (`lastCreatedAt` + ".." + time.Now()...`)?

Since the results are sorted by `created_at` descending, you'd always get the newest results, hence we need to scope the query "back in time" instead of always getting the latest values.
To clarify more:
1. `>=ParsedTimeSince` will give us the most recent 1000 workflow runs
2. On each loop we save the 1000th result `CreatedAt` then do `ParsedTimeSince...CreatedAt` which will give us the next 1000

This is all assuming the query has more than 1000 result (otherwise we won't this code).

If we could do `lastCreatedAt..time.Now()` we would have been able to use `lastCreatedAt` as state and not need this option

Let me write a test for this to make it clear

Reply:
> Since the results are sorted by `created_at` descending
yep, now that makes sense, no need for extra test, just doc it in the comments

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/17539/commits/81782372d3871945f779486d97a3b767cfe0a4a6